src/controllers/profile.js

```diff
@@ -0,0 +1,47 @@
+const bcrypt = require('bcrypt');
+const { validationResult } = require('express-validator');
+const ErrorResponse = require('../utils/errorResponse');
+const User = require('../models/user');
+
+exports.getMyProfile = async (req, res) => {
+  res.json({ success: true, data: req.user });
+
+  // This is an example of a successful response
+};
+exports.updateProfile = async (req, res, next) => {
+  const validationErrors = validationResult(req);
+  if (!validationErrors.isEmpty()) {
+    return res
+      .status(400)
+      .json({ success: false, errors: validationErrors.array() });
+  }
+
+  let hashedPassword;
+
+  if (req.body.password) {
+    hashedPassword = await bcrypt.hash(req.body.password, 10);
+  }
```

Comment:
Great work hashing the password, however you don't need to create outer variable and reassign it, instead, do 
```javascript
    req.body.password = await bcrypt.hash(req.body.password, 10);
```

However, there is another thing you need to do, you must get the `oldPassword` and `newPassword`, both, from the body, and verify if the `oldPassword` is correct, then update it.

Reply:
done